component/src/main/java/org/wso2/carbon/identity/authenticator/nuxeo/NuxeoAuthenticator.java

```diff
@@ -0,0 +1,386 @@
+            String userInfoUrl = getUserInfoEndpoint(authenticatorProperties);
+            OAuthAuthzResponse authorizationResponse = OAuthAuthzResponse.oauthCodeAuthzResponse(request);
+            String code = authorizationResponse.getCode();
+            AuthenticatedUser authenticatedUserObj;
+            Map<ClaimMapping, String> claims;
+            OAuthClientRequest accessRequest = getAccessRequest(tokenEndPoint, clientId, code, clientSecret,
+                    callbackUrl);
+            OAuthClient oAuthClient = new OAuthClient(new URLConnectionClient());
+            OAuthClientResponse oAuthResponse = getResponseOauth(oAuthClient, accessRequest);
+            String accessToken = oAuthResponse.getParam(OIDCAuthenticatorConstants.ACCESS_TOKEN);
+            String loggedInUserInfo = getUserInfoJson(userInfoUrl, accessToken);
+            if (loggedInUserInfo.isEmpty()) {
+                throw new AuthenticationFailedException("Getting empty or null value for logged in user info from " +
+                        "a user info endpoint: " + userInfoUrl);
+            }
+            Object loggedInUserID = JSONUtils.parseJSON(loggedInUserInfo).get(NuxeoAuthenticatorConstants.
+                    LOGGED_IN_USER_IDENTIFIER);
+            claims = mapUserInfo(loggedInUserInfo);
+            authenticatedUserObj = AuthenticatedUser.createFederateAuthenticatedUserFromSubjectIdentifier(
+                    String.valueOf(loggedInUserID));
+            authenticatedUserObj.setAuthenticatedSubjectIdentifier(String.valueOf(loggedInUserID));
+            authenticatedUserObj.setUserAttributes(claims);
+            context.setSubject(authenticatedUserObj);
+        } catch (OAuthProblemException e) {
+            throw new AuthenticationFailedException("Exception while getting the oAuth code from HttpServletRequest "
+                    + "for the client: " + clientId + ". Check whether you used the client_Secret value that " +
+                    "associate with clien_ID: " + clientId, e);
+        }
+    }
+
+    /**
+     * Get the OAuth response for access token.
+     *
+     * @param oAuthClient   the OAuthClient.
+     * @param accessRequest the AccessRequest.
+     * @return Response for access token from service provider.
+     */
+    private OAuthClientResponse getResponseOauth(OAuthClient oAuthClient, OAuthClientRequest accessRequest)
+            throws AuthenticationFailedException {
+
+        OAuthClientResponse oAuthResponse;
+        try {
+            oAuthResponse = oAuthClient.accessToken(accessRequest);
+        } catch (OAuthProblemException e) {
+            if (log.isDebugEnabled()) {
+                log.debug("Exception occurred from the invalid client_ID or Client_Secret: "
+                        + accessRequest.getBody() + " while getting the OAuth response for access token.");
+            }
+            throw new AuthenticationFailedException("Exception occurred from the invalid client_ID or Client_Secret " +
```

Comment:
Can we be sure that this error happens only because of invalid Client ID , secret ?